mne/morph.py

```diff
@@ -0,0 +1,1443 @@
+    transformation can be applied. Affine transformations are computed
+    based on the mutual information. This metric relates structural changes
+    in image intensity values. Because different still brains expose high
+    structural similarities this method works quite well to relate
+    corresponding features [1]_.
+    The nonlinear transformations will be performed as Symmetric
+    Diffeomorphic Registration using the cross-correlation metric [2]_.
+
+    References
+    ----------
+    .. [1] Mattes, D., Haynor, D. R., Vesselle, H., Lewellen, T. K., &
+    Eubank, W. (2003). PET-CT image registration in the chest using
+    free-form deformations. IEEE transactions on medical imaging, 22(1),
+    120-128.
+
+    .. [2] Avants, B. B., Epstein, C. L., Grossman, M., & Gee, J. C. (2009).
+    Symmetric Diffeomorphic Image Registration with Cross- Correlation:
+    Evaluating Automated Labeling of Elderly and Neurodegenerative Brain,
+    12(1), 26-41. Asymmetry. Journal of Cognitive Neuroscience 25(9),
+    1477-1492, 2013.
+    """
+    import nibabel as nib
+    from dipy.align import imaffine, imwarp, metrics, transforms
+    from dipy.align.reslice import reslice
+
+    logger.info('Computing nonlinear Symmetric Diffeomorphic Registration...')
+
+    morph_params = dict()
+
+    # use voxel size of mri_from
+    if spacing is None:
+        spacing = mri_from.header.get_zooms()[:3]
+
+    # use iso voxel size
+    if isinstance(spacing, (int, float)):
+        spacing = (float(spacing),) * 3
+
+    # reslice mri_from
+    mri_from_res, mri_from_res_affine = reslice(
+        mri_from.get_data(),
+        mri_from.affine,
+        mri_from.header.get_zooms()[:3],
+        spacing)
+
+    with warnings.catch_warnings(record=True):  # nibabel<->numpy warning
+        mri_from = nib.Nifti1Image(mri_from_res, mri_from_res_affine)
+
+    # reslice mri_to
+    mri_to_res, mri_to_res_affine = reslice(
+        mri_to.get_data(),
+        mri_to.affine,
+        mri_to.header.get_zooms()[:3],
+        spacing)
+
+    with warnings.catch_warnings(record=True):  # nibabel<->numpy warning
+        mri_to = nib.Nifti1Image(mri_to_res, mri_to_res_affine)
+
+    # get mri_to to world transform
+    mri_to_grid2world = mri_to.affine
+
+    # output mri_to as ndarray
+    mri_to = mri_to.dataobj[:, :, :]
+
+    # normalize values
+    mri_to = mri_to.astype('float') / mri_to.max()
+
+    # get mri_from to world transform
+    mri_from_grid2world = mri_from.affine
+
+    # output mri_from as ndarray
+    mri_from = mri_from.dataobj[:, :, :]
+
+    # normalize values
+    mri_from = mri_from.astype('float') / mri_from.max()
+
+    # compute center of mass
+    c_of_mass = imaffine.transform_centers_of_mass(mri_to, mri_to_grid2world,
+                                                   mri_from,
+                                                   mri_from_grid2world)
+
+    nbins = 32
```

Comment:
why is this hardcoded like this? 
if this has to be the case I would do this:

`imaffine.MutualInformationMetric(nbins=32, whatever=None)`